DependencyInjection/Configuration.php

```diff
@@ -73,6 +73,106 @@ public function getConfigTreeBuilder()
             ->end()
         ;
 
+        $this->addResourceSection($rootNode);
+
         return $treeBuilder;
     }
+
+    /**
+     * addResourceSection
+     *
+     * @param Node $rootNode
+     * @access private
+     * @return void
+     */
+    private function addResourceSection($rootNode)
+    {
+        $rootNode
+            ->fixXmlConfig('resource')
+            ->children()
+                ->arrayNode('resources')
+                    ->prototype('array')
+                        ->children()
+                            ->scalarNode('resource_class')
+                                ->cannotBeEmpty()
+                                ->defaultValue('Dunglas\ApiBundle\Api\Resource')
+                                ->info('The resource class name')
+                            ->end()
+                            ->scalarNode('entry_class')->cannotBeEmpty()->isRequired()->info('The entity class name')->end()
+                            ->scalarNode('short_name')->info('The resource short name')->end()
+
+                            // operations
+                            ->arrayNode('collection_operations')
+                                ->info('The collections operation to allow')
+                                    ->prototype('scalar')->end()
+                                    ->defaultValue(['GET', 'POST'])
+                            ->end()
+                            ->arrayNode('item_operations')
+                                ->info('The items operation to allow')
+                                    ->prototype('scalar')->end()
+                                    ->defaultValue(['GET', 'PUT', 'DELETE'])
+                            ->end()
+
+                            // custom operations
+                            ->arrayNode('collection_custom_operations')
+                                ->info('The custom collections operation to add')
+                                ->prototype('array')
+                                    ->children()
+                                        ->arrayNode('methods')
+                                            ->prototype('scalar')->end()
+                                        ->end()
+                                        ->scalarNode('path')->defaultNull()->end()
+                                        ->scalarNode('route')->defaultNull()->end()
+                                        ->scalarNode('controller')->isRequired()->end()
+                                        ->variableNode('context')->defaultValue(['hydra:title' => null])->end()
+                                    ->end()
+                                ->end()
+                            ->end()
+
+                            ->arrayNode('item_custom_operations')
```

Comment:
Custom operation should allow defining operation by its service id.
However, using the Symfony config component, it's not trivial to expose such a configuration. I had to play a lot with normalizations.